stubs/regex/regex/_regex.pyi

```diff
@@ -0,0 +1,185 @@
+from _typeshed import Self
+from typing import Any, AnyStr, Callable, Generic, Mapping, TypeVar, overload
+
+_T = TypeVar("_T")
+
+class Pattern(Generic[AnyStr]):
```

Comment:
`typing.pyi` marks some classes with `@final`, meaning that they can't be subclassed. This should be done here too:

```
>>> type(regex.compile('a'))
<class '_regex.Pattern'>
>>> class Foo(type(regex.compile('a'))): pass
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: type '_regex.Pattern' is not an acceptable base type
```